### PR TITLE
feat(issues): iteration field, search by iteration

### PR DESCRIPTION
**Pull request by lizalc**
Add missing Iteration field to Issue struct and support searching by
iteration ID.

Add tests for new features.

---

I've been using these changes for a work project for a few days now without issue.

### CODE REVIEW

**Review by svanharmelen — CHANGES_REQUESTED** [review]
Looks good, just a few small style comments...

File: issues_test.go [hunk 1]
```diff
@@ -324,6 +324,60 @@ func TestListIssuesSearchInDescription(t *testing.T) {
 		t.Errorf("Issues.ListIssues returned %+v, want %+v", issues, want)
 	}
 }
+func TestListIssuesSearchByIterationID(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (lizalc) [reply issues_test.go]: Done

File: issues_test.go [hunk 2]
```diff
@@ -387,6 +495,60 @@ func TestListGroupIssues(t *testing.T) {
 		t.Errorf("Issues.ListGroupIssues returned %+v, want %+v", issues, want)
 	}
 }
+func TestListGroupIssuesSearchByIterationID(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (lizalc) [reply issues_test.go]: Done

**Review by svanharmelen — APPROVED** [review]
Thanks 👍🏻

**Pull request merged**